src/main.rs

```diff
@@ -46,7 +36,20 @@ async fn main() -> error::Result<()> {
         panic!("You must enable at least one provider.");
     }
 
-    let store: HashMap<String, Client> = HashMap::new();
+    let database_url = config
+        .database_url
+        .as_deref()
+        .expect("database url is missing");
+
+    let store = PgPoolOptions::new()
+        .max_connections(5)
+        .connect(database_url)
+        .await?;
+
+    // Run database migrations. `./migrations` is the path to migrations, relative to the root dir (the directory
+    // containing `Cargo.toml`).
+    sqlx::migrate!("./migrations").run(&store).await?;
```

Comment:
what happens if we run multiple instances of the app?

Reply:
It has a migrations table, it will ignore as long as they are in-sync

Reply:
> what happens if we run multiple instances of the app?

There's a [locking mechanism](https://github.com/launchbadge/sqlx/blob/main/sqlx-core/src/migrate/migrator.rs#L70-L81) (enabled by default), which should prevent race conditions.